tests/uart_tests.py

```diff
@@ -0,0 +1,149 @@
+import threading
+import unittest, sys, os
+from supernovacontroller.sequential import SupernovaDevice
+from BinhoSupernova.commands.definitions import UartControllerBaudRate as baud, UartControllerParity as parity, UartControllerDataSize as dataSize, UartControllerStopBit as stopBit
+
+from supernovacontroller.sequential.uart import SupernovaUARTBlockingInterface
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+from binhosimulators import BinhoSupernovaSimulator
+
+# TODO @BMC2-1512 Update the simulator to support the execution/usage via SNController 
+class TestSupernovaControllerUART(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Initializes the testing class. Determines whether to use the simulator or real device
+        based on the "USE_REAL_DEVICE" environment variable. Default is to use the simulator.
+        """
+        cls.use_simulator = False
+
+    def setUp(self):
+        self.device = SupernovaDevice()
+        if self.use_simulator:
+            self.device.driver = BinhoSupernovaSimulator()
+
+    def test_uart_echo_message_real_device(self):
+        if self.use_simulator:
+            self.skipTest("For real device only")
+
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, _) = uart.set_bus_voltage(3300)
+        uart.init_bus()
+
+        data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+        (success, _) = uart.send(data)
+        (successReceive, response) = uart.wait_for_notification(1)
+
+        self.assertEqual(success, True)
+        self.assertTupleEqual((successReceive, response), (True, data), f"Failed echo: {response}")
+
+        self.device.close()
+
+    def test_set_bus_voltage(self):
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, _) = uart.set_bus_voltage(3300)
+
+        self.assertEqual(success, True)
+
+        self.device.close()
+
+    def test_wait_timeout(self):
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        uart.set_bus_voltage(3300)
+        uart.init_bus()
+
+        threading.Timer(5, lambda: self.fail("Timeout was exceeded"))
+        (success, response) = uart.wait_for_notification(4)
+        self.assertTupleEqual((False, "Timeout occurred while waiting for the UART receive notification"), (success, response), f"Failed non-timeout response: {response}")
```

Comment:
I know this is working but according [this](https://docs.python.org/es/3.9/library/threading.html#timer-objects) we should use .start() for threading.Timer(). Having this into account, the test can be something like this:

```
# Start the timer that will fail the test after 5 seconds
        timer = threading.Timer(5, lambda: self.fail("Timeout was exceeded"))
        timer.start()

        try:
            success, response = uart.wait_for_notification(4)
            self.assertTupleEqual(
                (False, "Timeout occurred while waiting for the UART receive notification"),
                (success, response),
                f"Failed non-timeout response: {response}"
            )
        finally:
            timer.cancel()
```